com.unity.multiplayer.mlapi/Runtime/Core/RPCQueue/RPCQueueHistoryFrame.cs

```diff
@@ -0,0 +1,226 @@
+using System.Collections.Generic;
+using MLAPI.Serialization.Pooled;
+
+namespace MLAPI
+{
+    /// <summary>
+    /// QueueHistoryFrame
+    /// All queued RPCs end up in a PooledBitStream within a QueueHistoryFrame instance.
+    /// </summary>
+    public class QueueHistoryFrame
+    {
+        public enum QueueFrameType
+        {
+            Inbound,
+            Outbound,
+        }
+
+        public uint TotalSize;
+
+        public PooledBitStream QueueStream;
+        public PooledBitWriter QueueWriter;
+        public PooledBitReader QueueReader;
+
+        public List<uint> QueueItemOffsets;
+
+        private int QueueItemOffsetIndex;
+        private FrameQueueItem CurrentQueueItem;
+        private readonly QueueFrameType queueFrameType;
+
+        long CurrentStreamSizeMark;
+
+        /// <summary>
+        /// GetQueueFrameType
+        /// Returns whether this is an inbound or outbound frame
+        /// </summary>
+        /// <returns></returns>
+        public QueueFrameType GetQueueFrameType()
+        {
+            return queueFrameType;
+        }
+
+        /// <summary>
+        /// MarkCurrentStreamSize
+        /// Marks the current size of the stream  (used primarily for sanity checks)
+        /// </summary>
+        public void MarkCurrentStreamPosition()
+        {
+            if (QueueStream != null)
+            {
+                CurrentStreamSizeMark = QueueStream.Position;
+            }
+            else
+            {
+                CurrentStreamSizeMark = 0;
+            }
+        }
+
+        public long GetCurrentMarkedPosition()
+        {
+            return CurrentStreamSizeMark;
+        }
+
+
+        /// <summary>
+        /// GetCurrentQueueItem
+        /// Internal method to get the current Queue Item from the stream at its current position
+        /// </summary>
+        /// <returns>FrameQueueItem</returns>
+        private FrameQueueItem GetCurrentQueueItem()
+        {
+            //Write the packed version of the queueItem to our current queue history buffer
+            CurrentQueueItem.QueueItemType = (RPCQueueManager.QueueItemType)QueueReader.ReadUInt16();
+            CurrentQueueItem.SendFlags = (Security.SecuritySendFlags)QueueReader.ReadUInt16();
+            QueueReader.ReadSingle();
```

Comment:
oh, is this `ReceiveTime` or what? should we just comment next to it to tell what it is?

Reply:
Adding to the new Jira Task!  :)